StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionPickerViewController.swift

```diff
@@ -113,18 +114,68 @@ class InstitutionPickerViewController: UIViewController {
 
     private func didSelectInstitution(_ institution: FinancialConnectionsInstitution) {
         searchBar.resignFirstResponder()
-        // clear search results
-        searchBar.text = ""
-        institutionTableView.load(
-            institutions: dataSource.featuredInstitutions,
-            isUserSearching: false
+
+        let showLoadingView: (Bool) -> Void = { [weak self] show in
+            guard let self else { return }
+            view.isUserInteractionEnabled = !show // prevent accidental taps
+            institutionTableView.showLoadingView(show, forInstitution: institution)
+        }
+
+        showLoadingView(true)
+        institutionTableView.showOverlayView(
+            true,
+            exceptForInstitution: institution
         )
-        delegate?.institutionPickerViewController(self, didSelect: institution)
+
+        dataSource.createAuthSession(institutionId: institution.id)
+            .observe { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case .success(let authSession):
+                    self.delegate?.institutionPickerViewController(
+                        self,
+                        didSelect: institution,
+                        authSession: authSession
+                    )
+
+                    if authSession.isOauthNonOptional {
+                        // oauth presents a sheet where we do not hide
+                        // the overlay until the sheet is dismissed
+                        observePartnerAuthDismissToHideOverlay()
+                    } else {
+                        hideOverlayView()
+                    }
+                case .failure(let error):
+                    // TODO(kgaidis): handle errors...
+                    print(error)
```

Comment:
this is intentionally left out for future PR

likely will require some heavier refactoring (a new way to present some errors)